website/content/en/blog/tracking-allocations.md

```diff
@@ -0,0 +1,50 @@
+description: Gaining insight into Vector's per-component memory usage.
+authors: ["arshiyasolei", "tobz"]
+date: "2022-12-13"
+badges:
+  type: announcement
+  domains: ["monitoring"]
+tags: ["allocation-tracing", "tracking allocations"]
+---
+
+We are excited to announce that Vector now has support for exposing per-component memory usage metrics. This work addresses an often-requested feature from users who want to understand how Vector uses memory, and what parts of their configurations are responsible for high memory usage.
+
+## Trying it out
+
+![vector top with allocation tracing](/img/blog/vector-top-allocation-tracking.png)
+
+To explore the feature, start a Vector instance with the `allocation-tracing` option, launch `vector top` and monitor your components!
+
+## What's new
+
+We provide the following new metrics: `component_allocated_bytes`, `component_allocated_bytes_total`, and `component_deallocated_bytes_total`.
+
+- `component_allocated_bytes` captures the current net allocations/deallocations.
+- `component_allocated_bytes_total` shows the accumulated total allocations.
+- `component_deallocated_bytes_total` shows the accumulated total deallocations.
+
+The following image visualizes the `component_allocated_bytes` metric of a sample Vector instance:
+![Visualization of the allocation tracking metrics via Datadog](/img/blog/vector-allocation-tracking-graph.png)
+
+## How it works
+
+Under the hood, Vector uses a custom memory allocator implementation that captures each allocation/deallocation and associates it with the currently-executing component. This work builds upon Vector's existing tracing functionality. Allocations that aren't associated with any components are tracked by a `root` component.
+
+## Using allocation tracing in production
+
+Currently, this feature is only supported on Unix builds of Vector. This may change in the future as we continue to improve allocation tracing.
+
+Additionally, there is a performance overhead imposed by allocation tracing when enabled. Vector runs a set of soak tests in CI to catch performance regressions during development. These soak tests emulate common Vector use cases, such as remapping events, or converting payloads from one event type to another, and so on.
+
+In our development and testing of this feature, we've observed ~20% reduction in throughput. This overhead may be acceptable for debugging purposes, but care should be taken before enabling it across your entire fleet.
+
+## Next steps
+
+We currently do not provide support for determining shared memory usage between components. For example, when a sink is batching events, and is configured to batch many events, you may observe high Vector memory usage. If you looked at the memory usage metrics, you would see most of it being attributed to components that either created the events (such as a source) or processed the events (such as any transforms that modified the event) rather than the sink which is batching the events.
```

Comment:
It feels weird to say that "we don't provide support", but maybe more than that, the example is talking about how we can't yet track how memory moves from one component to another in a logical fashion... whereas this wording is talking about "shared memory usage".

Nothing is being shared or double-counted, etc. I'll reiterate that this could stand to be reworded to focus on the notion that the allocation tracing currently doesn't fully track memory that moves from one component to another.

For reference, the first sentence of my original suggestion for rewriting this area:

> We currently only track memory usage in a way that shows which component is responsible for allocating memory, but not how that memory moves around Vector during normal operation.

Reply:
Reworded it to mention ownership instead